jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2962,15 +2962,7 @@ public static final  MethodHandle dropArgumentsToMatch(MethodHandle originalHand
 				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0673")); //$NON-NLS-1$
 			}
 		}
-
-		/* check if sublist match */
-		for (int i = skippedArgumentCount; i < ptypes.length; i++) {
-			if (ptypes[i] != valueTypesCopy[i + location - skippedArgumentCount]) {
-				/*[MSG "K0674", "Original handle types does not match given types at location"]*/
-				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0674")); //$NON-NLS-1$
-			}
-		}
-
+		
```

Comment:
Would something like this work with your changes?

```
		MethodHandle add2 = lookup.findStatic(Math.class, "addExact", methodType(int.class, int.class, int.class));
		MethodHandle modadd2 = dropArgumentsToMatch(add2, 1, MethodType.methodType(void.class, String.class, int.class, String.class, int.class, long.class, String.class).parameterList(), 5);
```

I would except an IllegalArgumentException to be thrown. Not sure if that will work if you remove the sublist match check.

Reply:
You're right the sublists should be the same. Let me see if I can find a better solution for this.

Reply:
The change to dropArgumentsToMatch was to fix a jck test in iteratedLoop. I've replaced the check in dropArgumentsToMatch and handled the error correctly in the iteratedLoop method. Just running personal build before I remove the WIP label.